### ISSUE TITLE
Exception at HTTP redirect 

### ISSUE THREAD

**Issue by moolen** [Unleash/unleash-client-java#113]
Some JAVA Clients use the unleash endpoint with **HTTP**: `http://toggle.example.org/api`. Today we did migrate the feature-toggle server and enforced a SSL redirect (308) to the very same HTTPS endpoint. The JAVA client throws an exception:
```
no.finn.unleash.UnleashException · Error fetching toggles from Unleash API - StatusCode: 308, Location: https://toggle.example.org/api/client/features
```
i think this is the code related to this:
https://github.com/Unleash/unleash-client-java/blob/12ed9c286264437efb752245c28f11c0eb01f2f2/src/main/java/no/finn/unleash/repository/HttpToggleFetcher.java#L45-L52

I expected that the unleash client would follow the redirects of `307` or `308` redirectsand fetch the toggles from there.

**Comment by ivarconr** [comment]
Thanks, I agree, Unleash client SDKs should follow redirects, also 302.

**Comment by michbeck100** [comment]
Are there any plans for the next release?

**Comment by ivarconr** [comment]
Released as part of `3.3.4`. Thanks for your contribution :fire:
